### ISSUE TITLE
Linkedin Auth Support

### ISSUE THREAD

**Issue by INF800** [wasp-lang/open-saas#19]
Hi, how easy is it to add Linkein auth? I don't see this in https://wasp-lang.dev/docs/auth/overview

**Comment by Elijas** [comment]
+1 for this

**Comment by Martinsos** [comment]
Yup, we don't have it yet, and you can't add it at the moment, but we will soon be adding more providers to Wasp (in the next 30 days most likely), including LinkedIn! Specifically, we will be integrating https://github.com/pilcrowOnPaper/arctic .

At some point in the future we might also add an option to provide a custom oAuth2 provider, although that will come later, as `arctic` should cover most of the popular ones.

**Comment by almereyda** [comment]
It will be interesting to test arctic against a self-hosted GitLab instance, and a subsequent implementation of an OIDC Discovery compatible client.

**Comment by LouisDeconinck** [comment]
> Yup, we don't have it yet, and you can't add it at the moment, but we will soon be adding more providers to Wasp (in the next 30 days most likely), including LinkedIn! Specifically, we will be integrating https://github.com/pilcrowOnPaper/arctic .
> 
> At some point in the future we might also add an option to provide a custom oAuth2 provider, although that will come later, as `arctic` should cover most of the popular ones.

Would be great to get these additional OAuth options.

**Comment by Martinsos** [comment]
> > Yup, we don't have it yet, and you can't add it at the moment, but we will soon be adding more providers to Wasp (in the next 30 days most likely), including LinkedIn! Specifically, we will be integrating https://github.com/pilcrowOnPaper/arctic .
> > At some point in the future we might also add an option to provide a custom oAuth2 provider, although that will come later, as `arctic` should cover most of the popular ones.
> 
> Would be great to get these additional OAuth options.

Uff time is passing by quickly! So we got much closer with this, we now have arctic integrated, but we need to add each provider manually. This is actually now quite straight-forward per provider, but still takes a bit of work.

I don't have an exact date, but it will be happening in the near future!

**Comment by giovanniborella** [comment]
@Martinsos If you can provide pointers, this might be a good first-issue? I'm looking for `Microsoft Entra ID` myself

**Comment by Martinsos** [comment]
> @Martinsos If you can provide pointers, this might be a good first-issue? I'm looking for `Microsoft Entra ID` myself

Unfortunately, I would call it simple enough to be a good first issue -> the process of doing it is not very "smart", but it does require a bunch of "boring" changes in a bunch of files, so would be tricky for first time contributor to figure out. We will likely make it simpler in the future but we are not there yet.
But good to know about "Microsoft Entra ID", we will see into that also!

**Comment by KochMaster** [comment]
𝕏 Auth would be really really great

Should be possible with arctic

**Comment by kotasudhakar** [comment]
+1 for this
